test/SparqlAntlrParserTestHelpers.h

```diff
@@ -224,7 +221,11 @@ class MultiVariantMatcher {
     const bool match = matcher_.MatchAndExplain(*elem, &elem_listener);
     *listener << "whose value " << testing::PrintToString(*elem)
               << (match ? " matches" : " doesn't match");
-    testing::internal::PrintIfNotEmpty(elem_listener.str(), listener->stream());
+    // First add our own Explanation and then that of the sub matcher.
+    auto subMatcherExplanation = elem_listener.str();
+    if (!subMatcherExplanation.empty()) {
+      *listener << subMatcherExplanation;
+    }
```

Comment:
```suggestion
    *listener << elem_listener.str();
```

1. appending empty strings should have no effect.
2. does *listener << elem_listener also work?
3. elem_listener should be called `elemListener` or maybe even `subMatcherListener` if that is what it is.